packages/rich-text/src/test/helpers/index.js

```diff
@@ -570,6 +573,49 @@ export const spec = [
 			text: '\ufffc',
 		},
 	},
+	{
+		description: 'should handle SVG',
+		html: '<svg><path d="M0,0 v2 h4"/></svg>',
+		NS_URI: 'http://www.w3.org/2000/svg',
+		selectTarget: ( body ) => body.querySelector( 'path' ),
+		createRange: ( element ) => ( {
+			startOffset: 0,
+			startContainer: element,
+			endOffset: 0,
+			endContainer: element,
+		} ),
+		startPath: [ 0, 0, 0 ],
+		endPath: [ 0, 0, 0 ],
+		record: {
+			start: 0,
+			end: 0,
+			formats: [ [ svg ] ],
+			replacements: [ path ],
```

Comment:
I don't like this. I don't understand why we need all this. Why not just reuse the existing logic we have for contenteditable false, and store innerHTML?

Reply:
Do we even need to think about namespaces if we just set innerHTML?

Reply:
Thanks for having another look! 

> Why not just reuse the existing logic we have for contenteditable false, and store innerHTML?

Outdated reason: this approach is three years old #24582 and back then I couldn't find innerHTML used in to-dom.js so I figured it was off-limits for some reason. Now, I will gladly explore this and very much appreciate the hint.

> Do we even need to think about namespaces if we just set innerHTML?

I would sure love it if that's the case 🤞.

Reply:
> Do we even need to think about namespaces if we just set innerHTML?

Right, namespaces are taken care of when setting `innerHTML`. I've learned that SVG in a format works already as long as it's nested inside. Still, I'd like if we can remove the need for a wrapper element. I've reduced this PR to be what I think is the minimal way to support that—let formats declare a namespace.